### PR TITLE
Python init fix

### PR DESCRIPTION
**Pull request by pretendWhale**
## Motivation and Context

In the production environment, the `markus_exam_matcher` package installs as `markus-exam-matcher` which leads the python initializer to set `Rails.application.config.scanner_enabled = false`, even though the package is installed and can be imported/invoked with `markus_exam_matcher`.

## Your Changes

**Description**:
This PR modifies the initializer to remove hyphens and underscores from package names in the various requirements files as well as the list of installed packages, which should allow both package names to be matched.

**Type of change** (select all that apply):

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

Manually tested with by restarting docker container, as well as using `binding.break` to test package names. NOTE: we should add unit tests for this (and other) initializers, but it requires moving the initialization script into a class. We should in the future make a separate PR which adds tests.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: config/initializers/python.rb [hunk 1]
```diff
@@ -42,6 +45,12 @@ def process_requirements(requirements_file)
              'Jupyter notebook rendering will be disabled. ' \
              "To enable notebook rendering run: #{pip} install -r #{Rails.root.join('requirements-jupyter.txt')}"
       end
+      unless (qr_requirements - installed).empty?
+        warn 'MARKUS WARNING: not all packages required to process scanned exams ' \
+             'are installed. Exam scanning will not be enabled. ' \
+             'To enable exam scanning  run: ' \
```
Comment (david-yz-liu) [inline config/initializers/python.rb:51]: @pretendWhale Please delete the double space

Comment (pretendWhale) [reply config/initializers/python.rb]: Did this for the `qr_requirements` warning. The `scanner_requirements` warning also has a double space - let me know if you think that's worth taking action on.

Comment (david-yz-liu) [reply config/initializers/python.rb]: @pretendWhale good catch, yeah please go ahead and fix it in both

Comment (pretendWhale) [reply config/initializers/python.rb]: @david-yz-liu Done!

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**